src/PathSimulator.cpp

```diff
@@ -287,6 +287,66 @@ void PathSimulator<DDPackage>::generateAlternatingSimulationPath(std::size_t sta
     setSimulationPath(components, true);
 }
 
+template<class DDPackage>
+void PathSimulator<DDPackage>::generateGatecostSimulationPath(std::size_t startingPoint, std::list<std::size_t> gateCosts) {
+    typename SimulationPath::Components components{};
+    components.reserve(CircuitSimulator<DDPackage>::qc->getNops());
+    std::size_t startElem = startingPoint;
+    components.emplace_back(startElem, startElem + 1);
+    std::size_t leftID   = startElem - 1;
+    std::size_t leftEnd  = 0;
```

Comment:
can be made `const`

Reply:
done and will be included in the latest commit